### PR TITLE
add a way to hide certain files and folders 

### PR DESCRIPTION
**Pull request by ahaenggli**
Faced the same issue as #76 on a Synology NAS. The folders are just called differently (@eaDir, #recycle).

### CODE REVIEW

**Comment by alcalbg** [comment]
Hi,

Thanks for the PR!

However, I cannot merge it since I have some concerns about the feature.

First, there are no tests to cover this, and that's why Codecov is reporting decreased coverage.

Second, I'm not sure what exactly are we trying to accomplish here:

1. If we're hiding files and folders on the back-end, someone could assume that the user cannot access those files anymore which is wrong. They could, for example, forge a rename or copy request against the hidden file and it would pop-up under a different name. For example, the secret.php file is hidden but if I forge a rename request, I could rename secret.php to ex_secret.php which will enable me to override this protection. I didn't actually try this but I can only assume that the same applies to zipping or batch downloading 'hidden' files. Again, someone could assume that hiding files is some kind of security feature.

2. If we're hiding files to remove the clutter, then this is a display-related feature that falls under the front-end.  Maybe there could be some kind of toggle button to hide/show hidden files as we can see on some desktop file managers. This way it will be clear to everyone that this is not a security feature but simply a user convenience.

Let me know your thoughts, any feedback is appreciated.

**Comment by ahaenggli** [comment]
You are welcome. I'm glad you check the PR's first ;-)
Thank you for sharing your thoughts.

I didn't even think that a user could see this as a kind of security feature.

In my case, the apache-user cannot delete the affected folders and files anyway, because he has too few rights ... Probably the Windows-OS recycle bin, mentioned in the issue above, cannot be deleted that easily either.

So, your second point makes more sense - it's absolutely front-end, not back-end. An additional toggle button is a nice idea to make it clear that it is only for user convenience. Or simply a checkbox 'show all files/folders'. The bottom right corner is still empty and it would probably not bother there. The button/checkbox could also be hidden if no file/folder filter was configured.

**Review by alcalbg — COMMENTED** [review]
This explanation seems too verbose for a config file, maybe we can move this to docs? https://docs.filegator.io/configuration/basic.html

Also, I'm not sure if we should leave default entries here.

'filter_entries' instead of 'filter_cwd_entries'?

File: configuration_sample.php [hunk 1]
```diff
@@ -19,6 +19,14 @@
         'editable' => ['.txt', '.css', '.js', '.ts', '.html', '.php', '.json', '.md'],
         'date_format' => 'YY/MM/DD hh:mm:ss', // see: https://momentjs.com/docs/#/displaying/format/
         'guest_redirection' => '', // useful for external auth adapters
+        'filter_cwd_entries' => ['Recycle.bin/', 'File System Information/', '.DS_Store', '@eaDir/', '#recycle/'],
```
Comment (alcalbg) [inline configuration_sample.php:22]: too verbose

File: frontend/views/Browser.vue [hunk 2]
```diff
@@ -144,11 +144,17 @@
...
-          <template slot="bottom-left">
+        <section class="is-flex is-justify-between">
+          <div>
             <span>{{ lang('Selected', checked.length, totalCount) }}</span>
-          </template>
-        </b-table>
+          </div>
+          <div v-if="(showAllEntries || hasFilteredEntries) ">
+            <input type="checkbox" id="checkbox" v-model="showAllEntries" @click="loadFiles">
+            <label for="checkbox"> {{ lang('show all files/folders') }}</label>
```
Comment (alcalbg) [inline frontend/views/Browser.vue:155]: Translations string should be something simpler, maybe 'show hidden'. 

TODO: add this to all translation files

**Comment by alcalbg** [comment]
Great, I like it!

**Comment by ahaenggli** [comment]
'filter_entries' would also be ok for me. 
Default entries could be a problem, but maybe you could use them as samples together with the explanation for the docs?

**Comment by alcalbg** [comment]
Thanks a lot!

I will fix a small bug when you select something and click on show hidden.

**Pull request merged**